Make RetireActualLRP remove the route from gorouter

vlast3k opened the pull request:
Thank you for submitting a pull request to the bbs repository!

We appreciate the contribution. To help us understand the context for your pull request, please fill out this template to the best of your ability.

## Please make sure to complete all of the following steps

1. Check the [Contributing document](https://github.com/cloudfoundry/diego-release/blob/develop/CONTRIBUTING.md) on how to sign the CLA and run tests.
1. Submit your PR to this repo.
1. [**Submit an accompanying PR Review Request**](https://github.com/cloudfoundry/diego-release/issues/new?assignees=&labels=&template=pr-review-request.md&title=%5BBBS+PR+REVIEW%5D%3A) referencing this PR so the Diego Team knows to review your pull request. 
* **Note: this PR will not be reviewed unless you submit the [PR Review Request](https://github.com/cloudfoundry/diego-release/issues/new?assignees=&labels=&template=pr-review-request.md&title=%5BBS+BPR+REVIEW%5D%3A)**.

***************************

## Please provide the following information:

### What is this change about?

> _Please describe._

### What problem it is trying to solve?

> _Please describe._

### What is the impact if the change is not made?

> _Please describe._

### How should this change be described in diego-release release notes?

> _Something brief that conveys the change. See [previous release notes](https://github.com/cloudfoundry/diego-release/releases) for examples._

### Please provide any contextual information.

> _Include any links to other PRs, stories, slack discussions, etc._

### Tag your pair, your PM, and/or team!

> _It's helpful to tag someone on your team or your team alias in case we need to follow up later._

Thank you!

mariash reviewed (changes requested):
Please see my comment in the issue

https://github.com/cloudfoundry/diego-release/issues/820#issuecomment-1721749046

vlast3k commented:
@mariash , i've changed the method, to just send an event, w/o removing the LRP. Indeed i did not recognize that in the other case the Actual LRP is not directly removed. 
I also added one additional unit test to validate it
